instrumentation/hibernate/hibernate-4.0/javaagent/src/test/groovy/SpringJpaTest.groovy

```diff
@@ -26,17 +25,33 @@ class SpringJpaTest extends AgentInstrumentationSpecification {
 
   def "test CRUD"() {
     setup:
+    def isHibernate4 = Version.getVersionString().startsWith("4.")
     def customer = new Customer("Bob", "Anonymous")
 
     expect:
     customer.id == null
     !repo.findAll().iterator().hasNext()
 
     assertTraces(1) {
-      trace(0, 1) {
+      trace(0, 4) {
         span(0) {
+          name "Session"
+          kind INTERNAL
+          hasNoParent()
+          attributes {
+          }
+        }
+        span(1) {
```

Comment:
Why do we need this span? Why do we create it at all?

Reply:
This span is created for the duration of hibernate session. What is a bit weird is that this span is used as a parent for all other hibernate spans.

Reply:
Wait, how long is "the duration of hibernate session" usually? Does it span multiple transactions/user requests or what is its lifecycle?

Reply:
@iNikem good questions, I opened #3552 to discuss

Reply:
See https://docs.jboss.org/hibernate/core/4.1/devguide/en-US/html/ch02.html#d5e626 typically there is one session per request but it is possible to have long sessions that span multiple requests.

Reply:
Some projects manage sessions manually, and sometimes fail miserably at this (been there, done that...)